### PR TITLE
Make the gainMap member of avifImage a pointer.

### PR DESCRIPTION
**Pull request by maryla-uc**
Allows adding members in the future without breaking the ABI.

### CODE REVIEW

File: src/write.c [hunk 1]
```diff
@@ -1208,8 +1208,10 @@ static avifResult avifValidateGrid(uint32_t gridCols,
     const avifImage * bottomRightCell = cellImages[cellCount - 1];
 #if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)
     if (validateGainMap) {
-        firstCell = firstCell->gainMap.image;
-        bottomRightCell = bottomRightCell->gainMap.image;
+        assert(firstCell->gainMap && firstCell->gainMap->image);
```
Comment (vrabaud) [inline src/write.c:1211]: Just wondering if that can never happen and whether  AVIF_CHECKRES should be used

**Review by vrabaud — APPROVED** [review]
Just wondering if there should be function that checks for gainMap and gainMap->image and returns NULL or the image to avoid doing the double condition every time. Then again, that would be more to maintain so no.

File: apps/avifgainmaputil/printmetadata_command.cc [hunk 2]
```diff
@@ -57,13 +57,13 @@ avifResult PrintMetadataCommand::Run() {
               << decoder->diag.error << ")\n";
     return result;
   }
-  if (!decoder->gainMapPresent) {
+  if (!decoder->gainMapPresent || decoder->image->gainMap == nullptr) {
```
[the remark this answers is not in the document]
Comment (maryla-uc) [reply apps/avifgainmaputil/printmetadata_command.cc]: In theory no. Changed to an assert.

File: src/write.c (hunk 1 above)
Comment (maryla-uc) [reply src/write.c:1211]: No the caller should only call the function with validateGainMap if there is a gain map. I think defensive programming is not useful in this instance.

**Pull request merged**